enigmatry-entry-blueprint-app/theming-setup.md

```diff
@@ -0,0 +1,120 @@
+
+The entry theme allows you to customize the styles of your application according to your needs. How to set up theming in your project, can be found in further sections.
+
+## Theme
+
+`_general.scss` file inside `styles/partials/vendors/libraries/entry/` folder is the place where it is located theme-related style configurations. The file contains imports for generator files for configuring theming, a Sass map in which you can define styles according to your needs, and a generator generates-from mixin, which passes a custom theme ( set of style configurations ) as an argument. If you don't change anything from the styles then the styling from the default theme will be applied.
+
+### Define custom theme
+
+Inside the `$custom-theme` map, you can define custom styles to suit specific preferences and needs.
+
+`$custom-theme` map includes 3 main sections:
+
+1) **general**
+- it contains styling for colors, density, typography, fonts, spacing, inputs, buttons
+2) **tables**
+- it contains styling for cells, rows, content
+3) **dialog**
+- it contains font size for the dialog title
+
+For more info about all these properties you can read [here](https://github.com/enigmatry/entry-angular-building-blocks/blob/master/libs/entry-components/configure-theming.md#1-custom-configuration).
+
+### Fonts
+
+There are 2 possibilities for using fonts in theming:
+
+1. Default Roboto font
+
+The theme comes with the default **Roboto** font family. Configuration is placed inside the `angular.json` file. The font is imported from the library's node modules.
+
+2. Custom fonts
+
+If you need a custom fonts, you can do it in the following way:
+
+1) put custom font files in the `assets/fonts/` folder of the project
+2) create typography.scss file inside the `partials/core/fonts` folder
+3) use the `@font-face` rule in `_typography.scss` file to define custom fonts with an absolute path
+
+```scss
+@font-face {
+  font: {
+    family: 'Helvetica';
+    weight: 400;
+  }
+  src: url('/assets/fonts/Helvetica.woff2') format('woff2'), url('/assets/fonts/Helvetica.woff') format('woff');
+}
+```
+4) remove the configuration from the assets part inside the `angular.json` file:
+
+```json
+"assets": [
+  {
+    "glob": "**/*",
+    "input": "./node_modules/@enigmatry/entry-components/assets/",
+    "output": "/assets/"
+  }
+]
+```
+### Adding Material classes
+
+Since Angular Material uses different typography levels, to be able to read values from the theme it's needed to add those typography-level cases to corresponding native elements. Check the full list [here](https://material.angular.io/guide/typography#typography-levels).
+
+We have added `.mat-body` class to the **body** element in `index.html` file in app root.
+
+We're overriding Angular Material, so it is important to add CSS classes for mixin that emit styles for native header elements. In another case, if don't provide those classes for custom components, typography configurations will not be applied.
```

Comment:
Why would anyone care we're overriding AM? We just need to say, please use this here and there for the right effect.